test/types/models.test.ts

```diff
@@ -211,4 +212,39 @@ function inheritance() {
 
 Project.createCollection({ expires: '5 seconds' });
 Project.createCollection({ expireAfterSeconds: 5 });
-expectError(Project.createCollection({ expireAfterSeconds: '5 seconds' }));
\ No newline at end of file
+expectError(Project.createCollection({ expireAfterSeconds: '5 seconds' }));
+
+export function autoTypedModel() {
+  const AutoTypedSchema = autoTypedSchema();
+  const AutoTypedModel = model('AutoTypeModel', AutoTypedSchema);
+
+  (async() => {
+  /* -------------------------------------------------------------------------- */
+  /*                            Model-functions-test                            */
+  /* -------------------------------------------------------------------------- */
+  // Create should works with arbitrary objects.
+    const randomObject = await AutoTypedModel.create({ unExistKey: 'unExistKey', description: 'st' });
+    expectType<string>(randomObject.unExistKey);
+    expectType<M0_0aAutoTypedSchemaType['schema']['userName']>(randomObject.userName);
+
+    const testDoc1 = await AutoTypedModel.create({ userName: 'M0_0a' });
+    expectType<M0_0aAutoTypedSchemaType['schema']['userName']>(testDoc1.userName);
```

Comment:
You use everywhere a prefix of 'M0_0a'. I woud prefer that you remove this and in the inferschematype.d.ts you write good header like

```
/*
 * @author Mohammad Ahmad (mohammad0-0ahmad) your@email.com
 */

```
If @vkarpov15  allows it, you could add yourself to the contributors list in the package.json.